src/field_theory/algebraic_closure.lean

```diff
@@ -0,0 +1,287 @@
+then root_of_splits _ ((splits_prod_iff _ $ λ (j : monic_irreducible k) _, j.2.2.ne_zero).1
+    (splitting_field.splits _) f hf)
+  (mt is_unit_iff_degree_eq_zero.2 f.2.2.not_unit)
+else 37
+
+theorem to_splitting_field_eval_X_self {s : finset (monic_irreducible k)} {f} (hf : f ∈ s) :
+  to_splitting_field k s (eval_X_self k f) = 0 :=
+by { rw [to_splitting_field, eval_X_self, ← alg_hom.coe_to_ring_hom, hom_eval₂,
+         alg_hom.coe_to_ring_hom, mv_polynomial.aeval_X, dif_pos hf,
+         ← algebra_map_eq, alg_hom.comp_algebra_map],
+  exact map_root_of_splits _ _ _ }
+
+theorem span_eval_ne_top : span_eval k ≠ ⊤ :=
+begin
+  rw [ideal.ne_top_iff_one, span_eval, ideal.span, ← set.image_univ, finsupp.mem_span_iff_total],
+  rintros ⟨v, _, hv⟩, replace hv := congr_arg (to_splitting_field k v.support) hv,
+  rw [alg_hom.map_one, finsupp.total_apply, finsupp.sum, alg_hom.map_sum, finset.sum_eq_zero] at hv,
+  { exact zero_ne_one hv },
+  intros j hj, rw [smul_eq_mul, alg_hom.map_mul, to_splitting_field_eval_X_self k hj, mul_zero]
+end
+
+/-- A random maximal ideal that contains `span_eval k` -/
+def max_ideal : ideal (mv_polynomial (monic_irreducible k) k) :=
+classical.some $ ideal.exists_le_maximal _ $ span_eval_ne_top k
+
+instance max_ideal.is_maximal : (max_ideal k).is_maximal :=
+(classical.some_spec $ ideal.exists_le_maximal _ $ span_eval_ne_top k).1
+
+theorem le_max_ideal : span_eval k ≤ max_ideal k :=
+(classical.some_spec $ ideal.exists_le_maximal _ $ span_eval_ne_top k).2
+
+/-- The first step of constructing `algebraic_closure`: adjoin a root of all monic polynomials -/
+def adjoin_monic : Type u :=
+(max_ideal k).quotient
+
+instance adjoin_monic.field : field (adjoin_monic k) :=
+ideal.quotient.field _
+
+instance adjoin_monic.inhabited : inhabited (adjoin_monic k) := ⟨37⟩
+
+/-- The canonical ring homomorphism to `adjoin_monic k`. -/
+def to_adjoin_monic : k →+* adjoin_monic k :=
+(ideal.quotient.mk _).comp C
+
+instance adjoin_monic.algebra : algebra k (adjoin_monic k) :=
+(to_adjoin_monic k).to_algebra
+
+theorem adjoin_monic.algebra_map : algebra_map k (adjoin_monic k) = (ideal.quotient.mk _).comp C :=
+rfl
+
+theorem adjoin_monic.is_integral (z : adjoin_monic k) : is_integral k z :=
+let ⟨p, hp⟩ := ideal.quotient.mk_surjective z in hp ▸
+mv_polynomial.induction_on p (λ x, is_integral_algebra_map) (λ p q, is_integral_add)
+  (λ p f ih, @is_integral_mul _ _ _ _ _ _ (ideal.quotient.mk _ _) ih ⟨f, f.2.1,
+    by { erw [polynomial.aeval_def, adjoin_monic.algebra_map, ← hom_eval₂,
+        ideal.quotient.eq_zero_iff_mem],
+      exact le_max_ideal k (ideal.subset_span ⟨f, rfl⟩) }⟩)
+
+theorem adjoin_monic.exists_root {f : polynomial k} (hfm : f.monic) (hfi : irreducible f) :
+  ∃ x : adjoin_monic k, f.eval₂ (to_adjoin_monic k) x = 0 :=
+⟨ideal.quotient.mk _ $ X (⟨f, hfm, hfi⟩ : monic_irreducible k),
+ by { rw [to_adjoin_monic, ← hom_eval₂, ideal.quotient.eq_zero_iff_mem],
+  exact le_max_ideal k (ideal.subset_span $ ⟨_, rfl⟩) }⟩
```

Comment:
Mathlib is not very consistent in how to align multi-line `by` blocks, but this seems to be more common:
```suggestion
 by { rw [to_adjoin_monic, ← hom_eval₂, ideal.quotient.eq_zero_iff_mem],
      exact le_max_ideal k (ideal.subset_span $ ⟨_, rfl⟩) }⟩
```

Reply:
Or just use tactic mode for the whole proof.